elab-util-extra/src/Deriving/SpecialiseData.idr

```diff
@@ -0,0 +1,1100 @@
+module Deriving.SpecialiseData
+
+import Control.Monad.Either
+import Control.Monad.Reader.Tuple
+import Data.SnocList
+import Data.DPair
+import Data.List1
+import Data.Vect
+import Data.List
+import Data.List.Quantifiers
+import Data.Either
+import Data.SortedMap
+import Data.SortedSet
+import Data.SortedMap.Dependent
+import Decidable.Equality
+import Derive.Prelude
+import Language.Mk
+import Language.Reflection.Expr
+import Language.Reflection.Logging
+import Language.Reflection.Syntax.Ops
+import Language.Reflection.TT
+import Language.Reflection.TTImp
+import Language.Reflection.Types.Extra
+import Language.Reflection.Unify
+import Language.Reflection.Util
+import Language.Reflection.VarSubst
+import Syntax.IHateParens
+
+%language ElabReflection
+
+%default total
+
+-------------------------------------
+--- SPECIALISATION TASK INTERFACE ---
+-------------------------------------
+
+||| Valid task lambda interface
+|||
+||| Auto-implemented by any Type or any function that returns Type.
+export
+interface TaskLambda (t : Type) where
+
+export
+TaskLambda Type
+
+export
+TaskLambda b => TaskLambda (a -> b)
+
+export
+TaskLambda b => TaskLambda (a => b)
+
+export
+TaskLambda b => TaskLambda ({_ : a} -> b)
+
+export
+TaskLambda b => TaskLambda ((0 _ : a) -> b)
+
+export
+TaskLambda b => TaskLambda ((0 _ : a) => b)
+
+export
+TaskLambda b => TaskLambda ({0 _ : a} -> b)
```

Comment:
Looks like this interface is used only in `specialisedData*`-functions, so I'd suggest to move it near to them